src/cpp/OpenWsmanClient.h

```diff
@@ -29,89 +30,147 @@ namespace WsmanClientNamespace
+			string Get(
+				const string &resourceUri,
+				const NameValuePairs *s = NULL) const;
 
 			// Update a resource.
-			string Put(const string &resourceUri, const string &content, const NameValuePairs *s = NULL) const;
+			string Put(
+				const string &resourceUri,
+				const string &content,
+				const NameValuePairs *s = NULL) const;
 
 			// Invokes a method and returns the results of the method call.
-			string Invoke(const string &resourceUri, const string &methodName, const string &content, const NameValuePairs *s = NULL) const;
+			string Invoke(
+				const string &resourceUri,
+				const string &methodName,
+				const WsmanOptions &options) const;
+			string Invoke(
+				const string &resourceUri,
+				const string &methodName,
+				const string &content,
+				const WsmanOptions &options) const;
+			string Invoke(
+				const string &resourceUri,
+				const string &methodName,
+				const string &content,
+				const NameValuePairs *s = NULL) const;
 
 			// Submit a subscription
-			string Subscribe(const string &resourceUri, const SubscribeInfo &info, string &subsContext) const;
+			string Subscribe(
+				const string &resourceUri,
+				const SubscribeInfo &info,
+				string &subsContext) const;
 
 			// Renew a subscription
-			string Renew(const string &resourceUri, const string &subsContext, float expire, const NameValuePairs *s = NULL) const;
+			string Renew(
+				const string &resourceUri,
+				const string &subsContext,
+				float expire,
+				const NameValuePairs *s = NULL) const;
 
 			// Terminate a subscription
-			void Unsubscribe(const string &resourceUri, const string &subsContext, const NameValuePairs *s = NULL) const;
+			void Unsubscribe(
+				const string &resourceUri,
+				const string &subsContext,
+				const NameValuePairs *s = NULL) const;
 
 			// Set auth method
-			void SetAuth(const char *auth_method = "digest");
+			void SetAuth(const string &auth_method = string("digest"));
 
 			// Set user name
-			void SetUserName(const char *user_name);
+			void SetUserName(const string &user_name);
 
 			// Set passsword
-			void SetPassword(const char *password);
+			void SetPassword(const string &password);
 
 			// Set encoding
-			void SetEncoding(const char *encoding);
-			// Set CIM namespace
-			void SetNamespace(const char *ns);
+			void SetEncoding(const string &encoding);
+
+			// Set/Get CIM namespace
+			void SetNamespace(const string &ns);
+			string GetNamespace() const;
 
-			void SetProxy(const char *proxy=NULL, const char *proxy_username=NULL, const char *proxy_password=NULL);
+			void SetProxy(
+				const string &proxy = string(),
+				const string &proxy_username = string(),
+				const string &proxy_password = string());
 #ifdef _WIN32
 			// Set client certificate params
-			void SetClientCert(const char *caOid=NULL, const char *caName=NULL, const bool localCert=false);
+			void SetClientCert(
+				const string &caOid = string(),
+				const string *caName = string(),
```

Comment:
This will not build on windows. Perhaps you meant this:
const string &caName = string(),